tests/viewer/test_event_db.cpp

```diff
@@ -0,0 +1,147 @@
+        _timestamp_field = std::make_shared<EventKlassField>("timestamp", "HT_TimestampNs", FieldTypeId::UINT64);
+
+        // Setup values for event's timestamps
+        _value1.f_UINT64 = 100;
+        _value2.f_UINT64 = 200;
+        _value3.f_UINT64 = 300;
+        _value4.f_UINT64 = 400;
+
+        // Setup events
+        _event1 = std::make_shared<Event>(_klass1);
+        _event2 = std::make_shared<Event>(_klass2);
+        _event3 = std::make_shared<Event>(_klass1);
+        _event4 = std::make_shared<Event>(_klass2);
+
+        // Set timestamps for events
+        _event1.get()->set_value(_timestamp_field.get(), _value1);
+        _event2.get()->set_value(_timestamp_field.get(), _value2);
+        _event3.get()->set_value(_timestamp_field.get(), _value3); 
+        _event4.get()->set_value(_timestamp_field.get(), _value4);
+    }
+
+    std::shared_ptr<EventKlass> _klass1;
+    std::shared_ptr<EventKlass> _klass2;
+    std::shared_ptr<EventKlassField> _timestamp_field;
+    std::shared_ptr<Event> _event1;
+    std::shared_ptr<Event> _event2;
+    std::shared_ptr<Event> _event3;
+    std::shared_ptr<Event> _event4;
+    FieldType _value1{};
+    FieldType _value2{};
+    FieldType _value3{};
+    FieldType _value4{};
+    EventDB event_db;
+};
+
+TEST_F(TestEventDB, CheckIfEventsAreKeptSorted)
+{
+    // Arrange
+    Query query;
+    HT_TimestampNs start_ts = (HT_TimestampNs)100;
+    HT_TimestampNs stop_ts = (HT_TimestampNs)200;
+
+    // Act
+    event_db.insert(*_event2);
+    event_db.insert(*_event1);
+    
+    // Assert
+    auto response = event_db.get_data(start_ts, stop_ts, query);
+    std::vector<Event> correct_response = {*_event1, *_event2};
+
+    ASSERT_EQ(response.size(), correct_response.size());
+    for (unsigned int i = 0; i < response.size(); ++i)
+    {
+        ASSERT_EQ(response[i].get().get_timestamp(), correct_response[i].get_timestamp());
+    }
+}
+
+TEST_F(TestEventDB, QuerySpecifcKlassId)
+{
+    // Arrange
+    Query query;
+    HT_TimestampNs start_ts = (HT_TimestampNs)100;
+    HT_TimestampNs stop_ts = (HT_TimestampNs)300;
+    query.klass_id = 1;
+
+    // Act
+    event_db.insert(*_event1);
+    event_db.insert(*_event2);
+    event_db.insert(*_event3);
+    auto response = event_db.get_data(start_ts, stop_ts, query);
+
+    // Assert
+    std::vector<Event> correct_response = {*_event1, *_event3};
+    ASSERT_EQ(response.size(), correct_response.size());
+    for (unsigned int i = 0; i < response.size(); ++i)
+    {
+        ASSERT_EQ(response[i].get().get_timestamp(), correct_response[i].get_timestamp());
+    }
+}
+
+
+TEST_F(TestEventDB, CheckLowerBoundConditions)
+{
+    // Arrange
+    Query query1, query2, query3;
+    HT_TimestampNs start_ts;
+    HT_TimestampNs stop_ts;
+
+    // Act
+    event_db.insert(*_event1);
+    event_db.insert(*_event2);
+    event_db.insert(*_event3);
+    event_db.insert(*_event4);
+
+    start_ts = (HT_TimestampNs)100;
```

Comment:
you could use `u` suffix, so explicit cast to `HT_TimestampNs` won't be necessary; i.e. `start_ts = 100u;`